src/components/BuyButton.tsx

```diff
@@ -15,79 +14,44 @@ import {
 import styled from "styled-components";
 
 import { zIndexes } from "./theme";
-import { allowedCountries } from "../lib/constants";
-// Make sure to call `loadStripe` outside of a component’s render to avoid
-// recreating the `Stripe` object on every render.
+import { useTranslation } from "react-i18next";
+import redirectToCheckout from "../helpers/redirectToCheckout";
 
 const Button = styled.button<
   SpaceProps & TypographyProps & BorderProps & BackgroundProps & LayoutProps
 >`
   width: auto;
   white-space: nowrap;
   z-index: ${zIndexes.inFront};
+  text-transform: uppercase;
   ${space};
   ${typography};
   ${border};
   ${background};
   ${layout};
 `;
 
-const stripePromise = loadStripe(`${process.env.REACT_APP_STRIPE_API_KEY}`);
-
-interface BuyButtonProps {
-  successUrl: string;
-  cancelUrl: string;
-}
-const BuyButton: React.FC<BuyButtonProps> = props => {
-  const { successUrl, cancelUrl } = props;
+const BuyButton: React.FC = () => {
   const [error, setError] = useState<string>();
-
-  const handleClick = async () => {
-    try {
-      // When the customer clicks on the button, redirect them to Checkout.
-      const stripe = await stripePromise;
-      if (!stripe) {
-        return null;
-      }
-
-      await stripe.redirectToCheckout({
-        lineItems: [
-          {
-            price: `${process.env.REACT_APP_STRIPE_PRICE_ID}`,
-            quantity: 1,
-          },
-        ],
-        mode: "payment",
-        successUrl,
-        cancelUrl,
-        shippingAddressCollection: {
-          allowedCountries: allowedCountries,
-        },
-      });
-      // If `redirectToCheckout` fails due to a browser or network
-      // error, display the localized error message to your customer
-      // using `error.message`.
-    } catch (error) {
-      setError(error.message);
-    }
-  };
+  const { t } = useTranslation();
 
   return (
     <Fragment>
       {error && <p>{error}</p>}
       <Button
         role="link"
-        onClick={handleClick}
+        onClick={redirectToCheckout(setError)}
```

Comment:
are you getting a typescript error here?

Reply:
nope its all good for me, sometimes the typescript compiler takes some time to catch up. so closing the file might work or if not, quitting vscode and reopening again should do the trick

![image](https://user-images.githubusercontent.com/11725595/100354918-28056c80-2ff1-11eb-8286-cd5f72ce4fab.png)